File: src/chipschallenge/blockreactions/ToggleWallTo.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package chipschallenge.blockreactions;

import chipschallenge.Block;
import chipschallenge.BlockContainer;
import chipschallenge.BlockContainerFullException;
import chipschallenge.Game;
import chipschallenge.GameLevel;

/**
 *
 * @author patrik
 */
public class ToggleWallTo extends BlockReaction {

    private ToggleWallTo() {}
    private static ToggleWallTo mInstance = null;
    public static synchronized ToggleWallTo getInstance() {
        if(mInstance == null)
            mInstance = new ToggleWallTo();
        return mInstance;
    }

    public void react(Block moving, Block standing) throws BlockContainerFullException {
        // No reaction
    }

    public boolean canMove(Block moving, Block standing) {
        return standing.isA(Block.Type.TOGGLEWALLOPEN);
    }

}

File: ChipsChallenge.Shared/Blockreactions/ToggleWallTo.cs
namespace ChipsChallenge.Shared.Blockreactions
{
    public class ToggleWallTo : NoSlipReaction
    {
        private static ToggleWallTo instance;

        private ToggleWallTo()
        {
        }

        public static ToggleWallTo Instance
        {
            get
            {
                lock (typeof(ToggleWallTo))
                {
                    return instance ?? (instance = new ToggleWallTo());
                }
            }
        }

        public override void React(Block moving, Block standing)
        {
        }

        public override bool canMove(Block moving, Block standing)
        {
            return standing.IsA(Block.Type.TOGGLEWALLOPEN);
        }
    }
}